src/tigerbeetle-java/src/main/java/com/tigerbeetle/Transfer.java

```diff
@@ -111,60 +184,131 @@ public void setPendingId(UUID pendingId) {
         }
     }
 
+    /**
+     *
+     * <p>
+     * Must always be interpreted as a positive integer.
+     *
+     * @return A 64-bits integer.
+     */
     public long getTimeout() {
         return timeout;
     }
 
+    /**
+     * @param timeout A 64-bits integer.
+     */
     public void setTimeout(long timeout) {
         this.timeout = timeout;
     }
 
+    /**
+     * Gets an identifier used to enforce transfers between the same ledger.
+     *
+     * @return A 32-bit integer.
+     */
     public int getLedger() {
         return ledger;
     }
 
+    /**
+     * Sets an identifier used to enforce transfers between the same ledger.
+     * <p>
+     * Must be non-zero.
+     * <p>
+     * Example: 1 for USD and 2 for EUR.
+     *
+     * @param ledger A 32-bit integer defined by the user.
+     */
     public void setLedger(int ledger) {
         this.ledger = ledger;
     }
 
-    public short getCode() {
+    /**
+     * Gets a reason for the transfer.
+     *
+     * @return A 16-bits unsigned integer.
+     */
+    public int getCode() {
         return code;
     }
 
+    /**
+     * Sets a reason for the transfer.
+     * <p>
+     * Must be non-zero.
+     * <p>
+     * Example: 1 for deposit, 2 for settlement.
+     *
+     * @param code A 16-bits unsigned integer defined by the user.
+     * @throws IllegalArgumentException if code is negative or greater than 65535.
+     */
     public void setCode(int code) {
         if (code < 0 || code > Character.MAX_VALUE)
-            throw new IllegalArgumentException("Code must be a unsigned 16 bits value");
+            throw new IllegalArgumentException("Code must be a 16-bits unsigned integer");
 
         this.code = (short) code;
     }
 
-    public short getFlags() {
+    /**
+     * Gets the behavior during transfers.
+     *
+     * @see com.tigerbeetle.TransferFlags
+     * @return A 16-bits unsigned integer bit mask.
```

Comment:
Can we turn these into meaningful Java enums or something in the client? Rather than requiring the user to do the work

Reply:
We had that but it was removed here: https://github.com/tigerbeetledb/tigerbeetle-java/commit/7ba7bbb0c6ced42bcf4a7d9f408041b884d16cc8#diff-49dd611e9452f59780634d2d13c834c281bf532a482553dab906dd928922f282

- Removed non-ergonomic enums AccountFlags and TransactionFlags in favor of a simple integer constant.
Maybe implementing a helper to combine the flags would be a better solution

Setting the flag is pretty straightforward since Java supports bitwise OR with `A | B`, but reading individual flags could use a helper.